clang/include/clang/Basic/DiagnosticDriverKinds.td

```diff
@@ -384,6 +384,9 @@ def err_drv_no_rdc_sycl_target_missing : Error<
   "linked binaries do not contain expected '%0' target; found targets: '%1', this is not supported with '-fno-sycl-rdc'">;
 def err_drv_fsycl_wrong_optimization_options : Error<
   "-fsycl-optimize-non-user-code option can be used only in conjunction with %0">;
+def warn_drv_fsycl_add_default_spec_consts_image_flag_in_non_AOT : Warning<
+  "-fsycl-add-default-spec-consts-image flag has an effect only in Ahead of Time Compilation mode (AOT).">,
+  InGroup<SyclTarget>;
```

Comment:
This will put this diagnostic under control of `-Wsycl-target`, this doesn't look to match (too generic?)

Reply:
Didn't think about it, thanks.
I am looking at available options in `DiagnosticGroup.td` and I can't choose one particular good option. Do you have any suggestions?

Also there is no a description of `SyclTarget`. However, from other warnings linked to `SyclTarget` it looks like this is related to some misuse of targets. Warning that I am adding is connected to wrong chosen target as well. Don't you think that it would be OK to leave it as is?

Reply:
Yes, I can see that association.  We can probably update the diagnostic to be more precise so it mentions that the use of `-fsycl-target` with an AOT value will address the warning.

Reply:
What do you think of the following option?
"This option is only useful if used in conjunction with Ahead of Time Compilation (-fsycl-target command line option)."

Reply:
Done